### ISSUE TITLE
Lint to prevent unhandled promised rejections

### ISSUE THREAD

**Issue by wincent** [liferay/liferay-frontend-projects#12]
Every promise should have a `catch()`, otherwise we risk creating log spew in the browser console that makes us look incompetent for a lack of error handling (and that would be a fair appraisal!).

Example of **incorrect** code:

```js
if (online) {
    client.connect(url).then(result => {
        dispatch({action: 'blah', result});
    });
}
```

(An unrelated nit with that code: if it is in a React component, it is probably missing an `isMounted()` check to guard against setting state of similar on an unmounted component.)

Examples of **correct** code:

```js
client.connect(url)
  .then(result => {
    dispatch({action: 'blah', result});
  })
  .catch(() => {
    // - gracefully recover; or:
    // - retry; or:
    // - log (maybe only when `process.env.NODE_ENV === 'development'`); or, at worst:
    // - swallow
  })
```

Note that it is OK not to `catch()` if you are returning your promise to somebody else: we assume that they do the right thing (even though we might not make an attempt to statically verify that):

```js
return client.connect(url);
```

For similar reasons (ie. that we assume the caller will deal with the problem) it's also ok not to `try`/`catch` when using `async`/`await` (currently we can only do that in the test environment, but [expect to be able to do it soon throughout portal](https://github.com/liferay/liferay-npm-tools/issues/338)):

```js
async function get(item) {
  // We expect you to do local `try`/`catch` if you want to handle the error locally,
  // but if you don't, we assume caller does it.
  const raw = await client.get(item);

  return JSON.parse(raw);
}
```

Basically the rationale of all this is to catch obvious mistakes, but other than that give people the freedom to structure their code as they wish without insisting that they litter it at every level with explicit error handling.
